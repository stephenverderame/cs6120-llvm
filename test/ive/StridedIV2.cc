// NOLINTBEGIN
#include <cstdio>
int main()
{
    int m = 20;
    for (int i = 5000; i >= 0; i -= m) {
        int p = m + i * 10;
        for (int j = i; j >= 0; j -= p) {
            int k = i + j * p;
            printf("%d\n", k);
        }
    }
}
// NOLINTEND