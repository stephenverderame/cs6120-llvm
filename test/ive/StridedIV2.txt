250105000
248108580
246120160
244139740
242167320
240202900
238246480
236298060
234357640
232425220
230500800
228584380
226675960
224775540
222883120
220998700
219122280
217253860
215393440
213541020
211696600
209860180
208031760
206211340
204398920
202594500
200798080
199009660
197229240
195456820
193692400
191935980
190187560
188447140
186714720
184990300
183273880
181565460
179865040
178172620
176488200
174811780
173143360
171482940
169830520
168186100
166549680
164921260
163300840
161688420
160084000
158487580
156899160
155318740
153746320
152181900
150625480
149077060
147536640
146004220
144479800
142963380
141454960
139954540
138462120
136977700
135501280
134032860
132572440
131120020
129675600
128239180
126810760
125390340
123977920
122573500
121177080
119788660
118408240
117035820
115671400
114314980
112966560
111626140
110293720
108969300
107652880
106344460
105044040
103751620
102467200
101190780
99922360
98661940
97409520
96165100
94928680
93700260
92479840
91267420
90063000
88866580
87678160
86497740
85325320
84160900
83004480
81856060
80715640
79583220
78458800
77342380
76233960
75133540
74041120
72956700
71880280
70811860
69751440
68699020
67654600
66618180
65589760
64569340
63556920
62552500
61556080
60567660
59587240
58614820
57650400
56693980
55745560
54805140
53872720
52948300
52031880
51123460
50223040
49330620
48446200
47569780
46701360
45840940
44988520
44144100
43307680
42479260
41658840
40846420
40042000
39245580
38457160
37676740
36904320
36139900
35383480
34635060
33894640
33162220
32437800
31721380
31012960
30312540
29620120
28935700
28259280
27590860
26930440
26278020
25633600
24997180
24368760
23748340
23135920
22531500
21935080
21346660
20766240
20193820
19629400
19072980
18524560
17984140
17451720
16927300
16410880
15902460
15402040
14909620
14425200
13948780
13480360
13019940
12567520
12123100
11686680
11258260
10837840
10425420
10021000
9624580
9236160
8855740
8483320
8118900
7762480
7414060
7073640
6741220
6416800
6100380
5791960
5491540
5199120
4914700
4638280
4369860
4109440
3857020
3612600
3376180
3147760
2927340
2714920
2510500
2314080
2125660
1945240
1772820
1608400
1451980
1303560
1163140
1030720
906300
789880
681460
581040
488620
404200
327780
259360
198940
146520
102100
65680
37260
16840
4420
0
