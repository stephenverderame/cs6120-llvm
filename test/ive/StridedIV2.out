Loop: 
7:                                                ; preds = %14, %4
  %.0 = phi i32 [ %.01, %4 ], [ %15, %14 ]
  %8 = icmp sge i32 %.0, 0
  br i1 %8, label %10, label %9

Derived IV: %11  = .0 * %6 
Derived IV: %12  = .0 * %6  + .01
Basic IV:   %15 = sub nsw i32 %.0, %6
Basic IV:   %.0 = phi i32 [ %.01, %4 ], [ %15, %14 ]
Loop: 
1:                                                ; preds = %26, %0
  %.01 = phi i32 [ 5000, %0 ], [ %27, %26 ]
  %2 = icmp sge i32 %.01, 0
  br i1 %2, label %4, label %3

Basic IV:   %27 = sub nsw i32 %.01, 20
Derived IV: %5  = .01 * i32 10
Derived IV: %6  = .01 * i32 10 + i32 20
Basic IV:   %.01 = phi i32 [ 5000, %0 ], [ %27, %26 ]
